front/src/providers/apollo/ApolloProvider.tsx

```diff
@@ -0,0 +1,19 @@
+import { ApolloProvider as ApolloProviderBase } from '@apollo/client';
+import { useRecoilState } from 'recoil';
+
+import { isMockModeState } from '@/auth/states/isMockModeState';
+
+import { apolloClient } from './apollo-client';
+import { mockClient } from './mock-client';
+
+export const ApolloProvider: React.FC<React.PropsWithChildren> = ({
```

Comment:
Could we try to keep the existing pattern which is : function ApolloProvider({ children }: React.PropsWithChildren)

Reply:
Why not using the official type for functional component, it enforce type and allow the use of `displayName` that can be useful during test.
```typescript
    type FC<P = {}> = FunctionComponent<P>;

    interface FunctionComponent<P = {}> {
        (props: P, context?: any): ReactNode;
        propTypes?: WeakValidationMap<P> | undefined;
        contextTypes?: ValidationMap<any> | undefined;
        defaultProps?: Partial<P> | undefined;
        displayName?: string | undefined;
    }
```

Reply:
Let's discuss it, I think consistency is more important overall. If we start using this notation, we should take this decision for the whole repository @magrinj @lucasbordeau